codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/awssdk/DafnyAwsSdkClientTypeConversionProtocol.java

```diff
@@ -763,4 +772,134 @@ func Error_FromDafny(err $L.Error)(error) {
         }
       );
   }
+
+  // Generates rest of the not visited shapes into a function
+  private void generateSerializerFunctions(final GenerationContext context, Set<ShapeId> alreadyVisited) {
```

Comment:
I think we should be able to optimize it to run along with the `ShapeVisitor`s . But since this runs outside of any production code - we are okay with this for now. Add a TODO.

Reply:
TODO was added in `generateSerializerFunctions` and `generateDeserializerFunctions` in aws sdk and local service's `TypeConversionProtocol`